src/compiler/scala/tools/nsc/backend/jvm/PostProcessorFrontendAccess.scala

```diff
@@ -145,25 +138,49 @@ object PostProcessorFrontendAccess {
       val optLogInline: Option[String] = s.YoptLogInline.valueSetByUser
       val optTrace: Option[String] = s.YoptTrace.valueSetByUser
     }
-
-    object backendReporting extends BackendReporting {
+    private lazy val localReporter = perRunLazy(this)(new ThreadLocal[BackendReporting])
+
+    override def withLocalReporter[T](reporter: BackendReporting) (fn : => T): T = {
+      val threadLocal = localReporter.get
+      val old = threadLocal.get()
+      threadLocal.set(reporter)
+      try fn finally
+        if (old eq null) threadLocal.remove() else threadLocal.set(old)
+    }
+    override def backendReporting : BackendReporting = {
+      val local = localReporter.get.get()
+      if (local eq null) directBackendReporting else local
+    }
+    object directBackendReporting extends BackendReporting {
+      //TODO backend reporting should not be locked, it should be buffered and flushed when we consume the result
```

Comment:
Can you elaborate what you mean by this comment?

Reply:
old comment. Originally synced, now buffered ( but not correctly buffered it seems from your other comments)